packages/beacon-node/src/chain/shufflingCache.ts

```diff
@@ -88,13 +93,18 @@ export class ShufflingCache implements IShufflingCache {
    * Insert a promise to make sure we don't regen state for the same shuffling.
    * Bound by MAX_SHUFFLING_PROMISE to make sure our node does not blow up.
    */
-  insertPromise(epoch: Epoch, decisionRoot: RootHex): void {
-    const promiseCount = Array.from(this.itemsByDecisionRootByEpoch.values())
-      .flatMap((innerMap) => Array.from(innerMap.values()))
-      .filter((item) => isPromiseCacheItem(item)).length;
-    if (promiseCount >= MAX_PROMISES) {
+  insertPromise(epoch: Epoch, decisionRoot: RootHex, stateRoot: RootHex): void {
```

Comment:
is `stateRoot` always state root? it's `REGEN_STATE_FOR_ATTESTATION_NO_STATE_ROOT_KNOWN` from `chain.ts` above
if yes we may rename to something like `stateRootOrReason: string`